pkg/table/tables/index.go

```diff
@@ -427,7 +434,9 @@ func (c *index) Delete(ctx table.MutateContext, txn kv.Transaction, indexedValue
 						if err != nil {
 							return err
 						}
-						if !h.Equal(oh) {
+						// The handle passed in may be a `PartitionHandle`,
+						// so we can't directly do comparation with them.
+						if !handleEqual(h, oh) {
```

Comment:
Although when `c.Global` = true h should be a partition handle, it is better to check `c.Global` here, we should compare pid when it's true.